src/lib.rs

```diff
@@ -983,6 +983,11 @@ fn valid_local_uri(user: &mut GooseUser, uri: &str) -> bool {
     }
 }
 
+/// Decodes the HTML. Currently it just decodes the encoded ampersand character.
+fn decode_html(html: &str) -> String {
+    html.replace("&amp;", "&")
```

Comment:
I'd be tempted to add a dependency here and to offload this logic, and to be sure we properly decode everything. What about instead using this? 
https://docs.rs/html-escape